spec/rspec/core/option_parser_spec.rb

```diff
@@ -81,28 +83,101 @@ def generate_help_text
 
     %w[ -v --version ].each do |option|
       describe option do
-        it "prints the version and exits" do
+        it "prints the version and returns a zero exit code" do
           parser = Parser.new([option])
-          expect(parser).to receive(:exit)
 
           expect {
-            parser.parse
+            options = parser.parse
+            exit_code = options[:runner].call
+            expect(exit_code).to eq(0)
           }.to output("#{RSpec::Core::Version::STRING}\n").to_stdout
         end
       end
     end
 
+    %w[ -X --drb ].each do |option|
+      describe option do
+        let(:parser) { Parser.new([option]) }
+        let(:options) { double(:options) }
+        let(:err) { double(:stderr, :puts => nil) }
+        let(:out) { double(:stderr, :puts => nil) }
+
+        it 'sets the `:drb` option to true' do
+          options = parser.parse
+
+          expect(options[:drb]).to be_truthy
+        end
+
+        context 'when a DRb server is running' do
+          it "builds a DRbRunner and runs the specs" do
+            drb_proxy = double(RSpec::Core::DRbRunner, :run => 0)
```

Comment:
This should ideally use `instance_double` as well -- we prefer to use them whenever possible since they provide stronger guarantees.